### DISCUSSION TITLE
is SSG for protected routes a good idea?

### DISCUSSION THREAD

**Discussion by lucksp** [vercel/next.js#28303] [Help] [answered]
I have a question about the dynamic URL as part of the return  & accessing auth token stored in the cookies
```
export const getStaticPaths = async () => {
  const resp = await get(url, { userToken } );
  const ids = resp.map(({data}) => {
    return {
       { params: { id: data.id } },
    }
  });
  return ({
    params: ids,
  });
};
```
This is a protected route.  I cannot access the userToken which is stored in the browser, since `getStaticPaths` is executed within the server.  Next requires the getStaticPaths …so i have to return something.  But without the token, it’s not possible.

What are some solutions to this (i would imagine) common problem?

I have also read something along the line of:
> If I have /[idUser]/how-much-i-make-in-a-year and you statically render that with someone's user id then that route has that user's protected data in it.

**Comment by nandorojo** [comment, marked as answer]
For protected routes, you should do your data fetching on the client after you know a user is successfully logged in. Or, you can use server side rendering instead of static for those pages. But the more typical approach is, show a skeleton, fetch data client-side, and then render (or redirect if they don't have access to the page).

**Reply by lucksp** [reply]
that is what I thought too...
but why is my dynamic route page complaining that it needs `getStaticPaths`.  This error link is provided:
https://nextjs.org/docs/messages/invalid-getstaticpaths-value

**Reply by nandorojo** [reply]
Because your code isn’t following the docs. It’s expecting an object with a `paths` field, which is an array  of objects with `params`. In any case, you should never include something like user ID in a static paths function. Consider this function “dumb”. It gets called when your app builds. Any user specific stuff would require server side rendering or just fetching on the client (recommended usually).
